llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -162,6 +162,14 @@ InstructionCost RISCVTTIImpl::getIntImmCostInst(unsigned Opcode, unsigned Idx,
     // split up large offsets in GEP into better parts than ConstantHoisting
     // can.
     return TTI::TCC_Free;
+  case Instruction::Store:
+    // If the address is a constant, use the materialization cost.
+    if (Idx == 1)
+      return getIntImmCost(Imm, Ty, CostKind);
```

Comment:
For addresses in [0, 2048), we can use `ld/sd a0, offset(zero)`.
(though I don't know if this is a common case)

Reply:
Good point! It also looks like for those cases the materialization cost already does the right thing and we don't hoist.

Reply:
Can you add a test for this case?

Reply:
Sure! Added.

Reply:
There are more things we should care about after diving into the code of `RISCVMatInt::getIntMatCost`:
* What if RVC exists?
* What about immediates that can be materialized via a single `LUI/SLLI/BSETI/...` like `0x80000000` (w/ or w/o RVC)? These immediates can't be folded into load/store offsets.
* Why we return the size of inst sequence as the cost w/o RVC, but the cost scales by 100 w/ RVC?

Reply:
> There are more things we should care about after diving into the code of `RISCVMatInt::getIntMatCost`:
> 
> * What if RVC exists?

From the `getIntImmCost` perspective, we should be able to just rely on `getIntMatCost` for that, no?

> * What about immediates that can be materialized via a single `LUI/SLLI/BSETI/...` like `0x80000000` (w/ or w/o RVC)? These immediates can't be folded into load/store offsets.

`getIntMatCost` does handle that as well.

> * Why we return the size of inst sequence as the cost w/o RVC, but the cost scales by 100 w/ RVC?

Correct me if I'm wrong, but I think `HasRVC` in `getInstSeqCost` is misleading. My understanding is that it's used to compute the cost of compression:
```
// If CompressionCost is true it will use a different cost calculation if RVC is
// enabled. This should be used to compare two different sequences to determine
// which is more compressible.
```
but the rest of the modeling through sequence generation does take into account RVC, just doesn't check for it to minimize differences in generated code:
```
    // NOTE: We don't check for C extension to minimize differences in generated
    // code.
```

`CompressionCost` is passed as `true` only in ISelLowering when the cost of two sequences is analyzed.

Reply:
I think I have misunderstood what the `RVC` means, thanks for the explanation!